lumigator/infra/mzai/helm/lumigator/templates/deployment.yaml

```diff
@@ -57,13 +58,14 @@ spec:
             - name: RAY_WORKER_GPUS
               value: {{ .Values.rayWorkerGPUs | quote }}
               # pragma: allowlist-nextline secret
-            {{- if .Values.mistralSecretName }}
-            - name: MISTRAL_API_KEY # kubectl create secret generic mistral-key --from-literal MISTRAL_API_KEY=value
+            - name: MISTRAL_API_KEY
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.mistralSecretName }}
-                  key: {{ .Values.mistralSecretKey }}
-            {{- end }}
+                  {{- /* Note that a one-line comments needs THIS precise layout {{- }} to avoid breaking flow... */}}
+                  {{- /* Note the lack of - at the if and end on each line */}}
```

Comment:
Is this really needed?

Reply:
Given the amount of time that I have spent fixing indentation and whitespacing in helm templates, I think it is 😆 I'd put it in some doc page too; maybe we can remove it from here at this point. But helm template indentation is imho tricky enough that a newcomer may find these comments helpful.

Reply:
Yep, agree with adding it in Confluence, but not in the code. I would rather have it as documentation, not as a comment.

Reply:
It might be cleaner to define this in `_helpers` and call it here

Reply:
To Mario's point, I think some of this could be captured in confluence,  https://mzai.atlassian.net/wiki/spaces/SD/pages/13271041/Helm+Best+Practices Specifically the callouts on using the dash `-` with `{{` and `}}` as it's more of a general helm-ism.

Reply:
> It might be cleaner to define this in `_helpers` and call it here

I'd usually agree, but the lack of proper functions in Helm and being a one line (however long) make me prefer having this piece here.

Reply:
I maintain that this type of information should be in Confluence not in the code

Reply:
Yes, it should be removed (if it's not, I just forgot, sorry in advance...). What I mean is...

> It might be cleaner to define this in _helpers and call it here

Which I'm *assuming* to be about https://github.com/mozilla-ai/lumigator/blob/20382c07ad36146529aab4b40a3d392f145e33b5/lumigator/infra/mzai/helm/lumigator/templates/deployment.yaml#L67. But maybe I got this wrong 🤔

Reply:
Yep, that one line probably should be a template helper since it's quite long. Within a given manifest it's better to stick with functions/pipelines, or calling named templates.

Reply:
Ok, an include should be fine here.